server/service/appconfig.go

```diff
@@ -743,7 +759,16 @@ func (svc *Service) validateMDM(
 			invalid.Append("macos_setup.enable_end_user_authentication",
 				`Couldn't enable macos_setup.enable_end_user_authentication because no IdP is configured for MDM features.`)
 		}
-		// TODO: Should we validate MDM configured on here too?
+	}
+
+	if mdm.MacOSSetup.EnableEndUserAuthentication != oldMdm.MacOSSetup.EnableEndUserAuthentication {
+		hasCustomConfigurationWebURL, err := svc.HasCustomSetupAssistantConfigurationWebURL(ctx, nil)
+		if err != nil {
+			return ctxerr.Wrap(ctx, err, "checking setup assistant configuration web url")
```

Comment:
Why return an error only here whereas the others are appending to `invalid`? I may have missed it but I didn't see the return error being checked anywhere.

Reply:
Sounds good! I agree it shouldn't be a bad request.